cleverhans/experimental/certification/nn.py

```diff
@@ -43,22 +47,70 @@ def __init__(self, net_weights, net_biases, net_layer_types):
     self.biases = []
     self.layer_types = []
     self.sizes = []
+    self.input_shapes = []
+    self.output_shapes = []
+    self.has_conv = False
+    if(input_shape is not None):
+      current_num_rows = input_shape[0]
+      current_num_columns = input_shape[1]
+      current_num_channels = input_shape[2]
+
     # Setting the sizes of the layers of the network
     # sizes[i] contains the size of x_i
     for i in range(self.num_hidden_layers):
       shape = np.shape(net_weights[i])
-      self.sizes.append(int(shape[1]))
+      # self.sizes.append(int(shape[1]))
```

Comment:
If this code if not needed then remove it completely instead of commenting.
Same for few lines below